libraries/WDT/src/WDT.cpp

```diff
@@ -0,0 +1,72 @@
+#include "WDT.h"
+
+// Constructor
+APM3_WDT::APM3_WDT()
+{
+    // The watchdog only runs off of the LFRC
+    am_hal_clkgen_control(AM_HAL_CLKGEN_CONTROL_LFRC_START, 0);
+    am_hal_wdt_init(&watchdogConfig);
+}
+
+// Configure the watchdog timer
+void APM3_WDT::configure(uint8_t clock, uint8_t interrupt, uint8_t reset)
+{
+  WDT->CFG_b.CLKSEL = clock; 
```

Comment:
Change  to setClock(clock) to take advantage of error checking

Reply:
So as to make the functions independent from one another, I simply added the same error checking if statement applied to setClock().